### PR TITLE
Update hebrew locale

### PR DESCRIPTION
**Pull request by ShlomoCode**
I am a native speaker

Fixes # (issue)

## Misc Checklist

- [ ] My change requires a documentation update on [Sparkle's website repository](https://github.com/sparkle-project/sparkle-project.github.io)
- [ ] My change requires changes to generate_appcast, generate_keys, or sign_update

## Testing

I tested and verified my change by using one or multiple of these methods:

- [ ] Sparkle Test App
- [ ] Unit Tests
- [ ] My own app
- [ ] Other (please specify)

(Describe all the cases that were tested)

macOS version tested: 14.5

### CODE REVIEW

**Comment by zorgiepoo** [comment]
GitHub won't allow me to merge this because it is still a draft.

**Comment by ShlomoCode** [comment]
Yes, I made a few quick fixes, in the next few days I'll finish it and get it out of draft mode

**Comment by ShlomoCode** [comment]
@zorgiepoo It's ready to merge

File: Sparkle/he.lproj/Sparkle.strings [hunk 1]
```diff
@@ -1,53 +1,152 @@
...
 
 /* No comment provided by engineer. */
-"Ready to Install" = "מוכן להתקנה!";
+"Quit %1$@, move it into your Applications folder, relaunch it from there and try again." = "צא מ-%1$@, העבר אותו לתיקיית היישומים שלך, הפעל אותו מחדש משם ונסה שוב.";
+
+/* No comment provided by engineer. */
+"Ready to Install" = "מוכן להתקנה";
+
+/* No comment provided by engineer. */
+"Should %1$@ automatically check for updates? You can always check for updates manually from the %1$@ menu." = "האם %1$@ צריך לבדוק באופן אוטומטי אם יש עדכונים? אתה תמיד יכול לחפש עדכונים באופן ידני מהתפריט של %1$@.";
```
Comment (zorgiepoo) [inline Sparkle/he.lproj/Sparkle.strings]: The first `%1$` should be `%1$@` I think.

Comment (zorgiepoo) [reply Sparkle/he.lproj/Sparkle.strings]: Also the second one might be wrong too.

File: Sparkle/he.lproj/Sparkle.strings [hunk 2]
```diff
@@ -1,53 +1,152 @@
...
+"Update Error!" = "העדכון נכשל!";
+
+/* No comment provided by engineer. */
+"Update Installed" = "העדכון הותקן בהצלחה";
+
+/* No comment provided by engineer. */
+"Updating %@" = "מעדכן את %@";
+
+/* No comment provided by engineer. */
+"Use Finder to copy %1$@ to the Applications folder, relaunch it from there, and try again." = "השתמש ב-Finder כדי להעתיק את %1$@ לתיקיית היישומים, הפעל אותו מחדש משם ונסה שוב.";
```
Comment (zorgiepoo) [inline Sparkle/he.lproj/Sparkle.strings:140]: Formatter token looks like it may be wrong here too.

File: Sparkle/he.lproj/Sparkle.strings [hunk 3]
```diff
@@ -1,53 +1,152 @@
...
+
+/* No comment provided by engineer. */
+"Version History" = "היסטוריית גרסאות ";
```
Comment (zorgiepoo) [inline Sparkle/he.lproj/Sparkle.strings]: Should there be an extra space at the end?

File: Sparkle/he.lproj/Sparkle.strings (hunk 1 above)
Comment (ShlomoCode) [reply Sparkle/he.lproj/Sparkle.strings]: It seems to me that it is wrong only in GitHub/VSCode's **display** which is LTR, and therefore causes a distorted display of mixed Hebrew (RTL) and English (LTR) text
When I copy the line to a div that is set to RTL, it displays properly: https://jsitor.com/WNpswH34x
CleanShot 2024-06-03 at 02 23 08@2x

File: Sparkle/he.lproj/Sparkle.strings (hunk 3 above)
Comment (ShlomoCode) [reply Sparkle/he.lproj/Sparkle.strings]: Thanks

**Comment by zorgiepoo** [comment]
The Hebrew localization is not currently enabled in SUUpdatePermissionPrompt.xib. It can be enabled in Xcode's File inspector for that file.

**Comment by ShlomoCode** [comment]
I have activated the SUUpdatePermissionPrompt file now

File: Sparkle/he.lproj/Sparkle.strings (hunk 2 above)
Comment (ShlomoCode) [reply Sparkle/he.lproj/Sparkle.strings:140]: Same as https://github.com/sparkle-project/Sparkle/pull/2573#discussion_r1623669653

File: Sparkle/he.lproj/Sparkle.strings (hunk 1 above)
Comment (zorgiepoo) [reply Sparkle/he.lproj/Sparkle.strings]: Screenshot 2024-06-02 at 9 07 03 PM

Is this how you expect it to look like? It's best to check how the strings look at runtime in an app. I reset the defaults of the Sparkle Test App with `defaults delete org.sparkle-project.SparkleTestApp` and launched the app twice to get the permission prompt.

Comment (ShlomoCode) [reply Sparkle/he.lproj/Sparkle.strings]: Yes it looks great
I have now also tested the all flow with Sparkle Test App and the format is valid, there are some minor RTL issues but I will file a separate issue on that

**Comment by zorgiepoo** [comment]
Thank you. I'll merge after the CI automation is done verifying the changes.

(One note: it's easier for me to review/track changes if you don't force push commits on the pull request branch. Also typically people create a separate branch when merging changes so they can keep their 2.x branch in sync. Edit: never mind, I see GitHub provides me of a way of comparing different revisions still. Oh well).

**Comment by ShlomoCode** [comment]
Would you like to wait for me to add RLM tags or should we do it in a new PR?

**Comment by zorgiepoo** [comment]
Let's separate it and wait until this one merges.

**Pull request merged**